src/Controller/QuizInstanceController.php

```diff
@@ -33,18 +34,17 @@ public function __construct(RendererInterface $renderer, QuizInstanceService $qu
 
     public function showCandidateQuizzes(Request $request, array $requestAttributes)
     {
```

Comment:
If you need the Session in the Controller inject it. (Consider using a wrapper over the session for Authentication). Services should expose their properties through getters and the Controller should not know that the service has a Session object. IMO you broke encapsulation.